content/en/docs/Monitoring Kyverno With Prometheus Metrics/_index.md

```diff
@@ -0,0 +1,55 @@
+---
+title: Monitoring Kyverno
+description: Monitor the activities associated with the Kyverno policies applied over your cluster with a good set of Prometheus-compliant metrics
+weight: 45
+---
+
+## Introduction
+
+As a cluster admistrator or just a normal dev associated with a Kubernetes cluster, it can certainly prove to be beneficial for you to have capabilities around monitoring the state of the Kyverno policies applied over your cluster. Things like tracking the policies, the changes associated with them, the activity associated with the incoming requests hitting them and the results associated with them can prove to be extremely useful as a part of cluster-observability compliance.
+
+Alongside this, providing a non-rigid granularity of monitoring the above targets from rule's level to policy's level to entire admission request's level would give you enough flexibility to extract a diversified set of insights from these metrics depending on your need or/and SLA requirements.
+
+## Installation and Setup
+
+Whenever you install Kyverno via helm, a service called `kyverno-svc` gets created as well inside the `kyverno` namespace and this service ends up exposing the metrics at its port no. 8000.
+
+```sh
+$ values.yaml
+
+...
+service:
+  port: 443
+  type: ClusterIP
+  # Only used if service.type is NodePort
+  nodePort:
+  ## Kyverno's metrics server will be exposed at this port
+  metricsPort: 8000
+  ## The Node's port which will allow access Kyverno's metrics at the host level. Only used if service.type is NodePort.
+  metricsNodePort: 8000
+  ## Provide any additional annotations which may be required. This can be used to
+  ## set the LoadBalancer service type to internal only.
+  ## ref: https://kubernetes.io/docs/concepts/services-networking/service/#internal-load-balancer
+  ##
+  annotations: {}
+...
+```
+
+By default, the service type is going to be `ClusterIP` meaning that the metrics would be only capable of being scraped by a Prometheus server sitting inside the cluster. <br>
+But speculatively, in majority of the cases, the Prometheus server would be kept outside the cluster as an isolated component. In those kinds of scenarios, you would want the `kyverno-svc` service to be publicly exposed so as to expose the metrics (available at port 8000) to your Prometheus server sitting outside the cluster.<br>
+
+To do so, either you can expose the `kyverno-svc` service as NodePort or LoadBalancer.
+That can be done via the `values.yaml` file itself which is provided at the time of helm installation.
+
+```sh
+...
+service:
+  type: NodePort
+  metricsPort: 8000
+  metricsNodePort: 8000
+...
+```
+
+The above configuration will allow Kyverno's metrics to be exposed at port 8000 of any host/node of the cluster.
```

Comment:
I would suggest we simplify this (line 38 - 53) to:

By default, Kyverno installs with service type `ClusterIP`. To expose the Kyverno service outside the cluster you can configure a different [service type](https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types), or use an Ingress (https://kubernetes.io/docs/concepts/services-networking/ingress/0/).

Reply:
So Jim, the way this helm chart is configured, it allows the user to expose the kyverno service of any type i.e. `ClusterIP` (default and only privately accessibly), `NodePort` (publicly accessible over a host's port), `LoadBalancer` (publicly accessible over a cloud-provisioned LoadBalancer).
So, that end is sorted.

Also, after this PR (https://github.com/kyverno/kyverno/commit/f1491fe6d3e813682c4efaafd29e0f6a16e2f94a), the metrics will be exposed as an entirely different service with the name `kyverno-svc-metrics`. Hence, the user can choose to publicly expose metrics (By configuring the `kyverno-svc-metrics` to be of the type NodePort or LoadBalancer) and still keep the original `kyverno-svc` private to the cluster as it's not needed to be exposed.

Reply:
I'll update the docs accordingly with this :)